editoast/src/views/timetable.rs

```diff
@@ -452,6 +440,63 @@ impl Conflict {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize, ToSchema)]
+pub struct PacedTrainOccurrenceId {
+    pub paced_train_id: i64,
+    pub index: u64,
+}
+
+impl Display for PacedTrainOccurrenceId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}#{}", self.paced_train_id, self.index)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum TrainId {
+    TrainSchedule(i64),
+    PacedTrainOccurrence(PacedTrainOccurrenceId),
+}
+
+impl FromStr for PacedTrainOccurrenceId {
+    type Err = String;
```

Comment:
```suggestion
    type Err = &'static str;
```

(not 100% sure though)